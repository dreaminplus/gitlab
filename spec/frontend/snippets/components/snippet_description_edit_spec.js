import SnippetDescriptionEdit from '~/snippets/components/snippet_description_edit.vue';
import { shallowMount } from '@vue/test-utils';

describe('Snippet Description Edit component', () => {
  let wrapper;
  let description = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.';
  const markdownPreviewPath = 'foo/';
  const markdownDocsPath = 'help/';

  function createComponent() {
    wrapper = shallowMount(SnippetDescriptionEdit, {
      propsData: {
        description,
        markdownPreviewPath,
        markdownDocsPath,
      },
    });
  }

  beforeEach(() => {
    createComponent();
  });

  afterEach(() => {
    wrapper.destroy();
  });

  describe('rendering', () => {
    it('matches the snapshot', () => {
      expect(wrapper.element).toMatchSnapshot();
    });

    it('renders the field expnanded when description exists', () => {
      expect(wrapper.find('.js-collapsed').classes('d-none')).toBe(true);
      expect(wrapper.find('.js-expanded').classes('d-none')).toBe(false);
    });

    it('renders the field collapsed if there is no description yet', () => {
      description = '';
      createComponent();

      expect(wrapper.find('.js-collapsed').classes('d-none')).toBe(false);
      expect(wrapper.find('.js-expanded').classes('d-none')).toBe(true);
    });
  });
});
