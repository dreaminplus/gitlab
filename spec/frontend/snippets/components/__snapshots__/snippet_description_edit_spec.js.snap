// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snippet Description Edit component rendering matches the snapshot 1`] = `
<div
  class="form-group js-description-input"
>
  <label>
    Description (optional)
  </label>
   
  <div
    class="js-collapsible-input"
  >
    <div
      class="js-collapsed d-none"
    >
      <gl-form-input-stub
        class="form-control"
        data-qa_selector="description_placeholder"
        placeholder="Optionally add a description about what your snippet does or how to use it..."
      />
    </div>
     
    <div
      class="js-expanded"
    >
      <markdown-field-stub
        addspacingclasses="true"
        canattachfile="true"
        enableautocomplete="true"
        helppagepath=""
        markdowndocspath="help/"
        markdownpreviewpath="foo/"
        note="[object Object]"
        quickactionsdocspath=""
        textareavalue=""
      >
        <textarea
          aria-label="Description"
          class="note-textarea js-gfm-input js-autosize markdown-area
          qa-description-textarea"
          data-supports-quick-actions="false"
          dir="auto"
          id="issue-description"
          placeholder="Write a comment or drag your files hereâ€¦"
        />
      </markdown-field-stub>
    </div>
  </div>
</div>
`;
